src/Amazon.Extensions.CognitoAuthentication/CognitoUserAuthentication.cs

```diff
@@ -431,6 +430,15 @@ public async Task<AuthFlowResponse> RespondToMfaAuthAsync(RespondToMfaRequest mf
         /// if one exists</returns>
         public async Task<AuthFlowResponse> RespondToMfaAuthAsync(RespondToMfaRequest mfaRequest, CancellationToken cancellationToken)
         {
+            if (mfaRequest == null)
+            {
+                throw new ArgumentNullException(nameof(mfaRequest));
+            }
+            if (mfaRequest.ChallengeNameType != ChallengeNameType.SMS_MFA && mfaRequest.ChallengeNameType != ChallengeNameType.SOFTWARE_TOKEN_MFA)
```

Comment:
Are these the only two types that are allowed in `RespondToAuthChallengeRequest`? If yes, is there a documentation page we can mention here in a comment?

Reply:
Logic of this method require only two of this values. There are call of [GetChallengeParamCodeName](https://github.com/DmitryProskurin/aws-sdk-net-extensions-cognito/blob/c166c6a9608268543cf43d71c134896fed0c8aa5/src/Amazon.Extensions.CognitoAuthentication/CognitoUserAuthentication.cs#L446) that return null value if argument was different from SMS_MFA or SOFTWARE_TOKEN_MFA. I don't know respond from [RespondToAuthChallengeAsync](https://github.com/DmitryProskurin/aws-sdk-net-extensions-cognito/blob/c166c6a9608268543cf43d71c134896fed0c8aa5/src/Amazon.Extensions.CognitoAuthentication/CognitoUserAuthentication.cs#L460), if ChallengeResponses will contain pair {null, "user-mfa-code"}.
I can write about it in comment at mfaRequest parameter.

Reply:
I can't found any documentation about mfa types different from sms or software token.

Reply:
Hi @DmitryProskurin, I would rather not change the logic here since we don't want to introduce a breaking change since the pair {null, "user-mfa-code"} might be valid. I believe that the change that you made in this method was intended to make debugging easier, and it isn't part of fixing the bug. If that is the case, you can revert the changes in method RespondToMfaAuthAsync.